#4414 - Interface Policy Questions

ninosamson opened the pull request:
## Changes
- [x] Adds interface policy questions for partner to student and supporting user 25/26 FT application
- [x] Added BCEA income amounts as keys to consolidated data
- [x] Minor fixes to hide panel labels in supporting user 25/26 form 

## Supporting User: 
![image](https://github.com/user-attachments/assets/84c20f21-f594-427d-84eb-a8ca242f5901)


## Student 25/26 FT: 
![image](https://github.com/user-attachments/assets/884dd879-55c0-4515-9770-a0b5d975cc04)

andrewsignori-aot commented on sources/packages/backend/workflow/src/workflow-definitions/load-assessment-consolidated-data.bpmn:
```diff
@@ -125,9 +125,11 @@
           <zeebe:header key="partner1HasEmploymentInsuranceBenefits" value="$.supportingUsers.Partner1.supportingData.hasEmploymentInsuranceBenefits" />
           <zeebe:header key="partner1HasFedralProvincialPDReceipt" value="$.supportingUsers.Partner1.supportingData.hasFedralProvincialPDReceipt" />
           <zeebe:header key="partner1HasTotalIncomeAssistance" value="$.supportingUsers.Partner1.supportingData.hasTotalIncomeAssistance" />
+          <zeebe:header key="partner1BCEAIncomeAssistanceAmount" value="$.supportingUsers.Partner1.supportingData.partnerBceaIncomeAssistanceAmount" />
```
Please rename to `partnerBCEAIncomeAssistanceAmount`.

andrewsignori-aot replied:
The idea was to follow what was already in the form. If the other variables in the same form.io are prefixed with `partner`, it should be ok, otherwise, we should keep the form.io definition pattern.

andrewsignori-aot commented on sources/packages/forms/src/form-definitions/sfaa2025-26-ft.json:
```diff
@@ -5368,6 +5368,76 @@
+                "onlyAvailableItems": true
+              },
+              "validateWhenHidden": false,
+              "key": "partnerHasBceaIncomeAssistance",
```
Please change the name to `partnerHasBCEAIncomeAssistance`.

ninosamson replied:
Updated.

ninosamson commented:
> Nice work, overall looks good, please take a look at the variable names. In general, if an acronym is in the middle of the sentence, it goes uppercase; if it is at the beginning, it goes lowercase. See examples below.
> 
> * bceaIncomeAssistanceAmount
> * studentDataPartnerBCEAIncomeAssistanceAmount

Thanks for the review @andrewsignori-aot. I have addressed comments.

bidyashish commented on sources/packages/backend/workflow/src/workflow-definitions/load-assessment-consolidated-data.bpmn:
```diff
@@ -125,9 +125,11 @@
           <zeebe:header key="partner1HasEmploymentInsuranceBenefits" value="$.supportingUsers.Partner1.supportingData.hasEmploymentInsuranceBenefits" />
           <zeebe:header key="partner1HasFedralProvincialPDReceipt" value="$.supportingUsers.Partner1.supportingData.hasFedralProvincialPDReceipt" />
           <zeebe:header key="partner1HasTotalIncomeAssistance" value="$.supportingUsers.Partner1.supportingData.hasTotalIncomeAssistance" />
+          <zeebe:header key="partner1BCEAIncomeAssistanceAmount" value="$.supportingUsers.Partner1.supportingData.bceaIncomeAssistanceAmount" />
```
`partner1BCEAIncomeAssistanceAmount` Does it need to prefix with `partner` only?

ninosamson replied:
This is the key for consolidated data - and follows the naming convention for other keys.

bidyashish commented on sources/packages/forms/src/form-definitions/sfaa2025-26-ft.json:
```diff
@@ -5368,6 +5368,76 @@
+                "onlyAvailableItems": true
+              },
+              "validateWhenHidden": false,
+              "key": "partnerHasBCEAIncomeAssistance",
```
Here it is just `partnerHasBCEAIncomeAssistance` and does not contains `partner1`

ninosamson replied:
Thanks Ashish! This key is for the formio question and does not need to have the "partner1" prefix.

bidyashish reviewed (commented):
LGTM

bidyashish reviewed (approved):
LGTM @ninosamson
